aten/src/ATen/core/ivalue.h

```diff
@@ -751,6 +751,38 @@ DEFINE_TO(c10::Device, toDevice)
 DEFINE_TO(at::ScalarType, toScalarType)
 DEFINE_TO(at::Layout, toLayout)
 
+template<typename Elem>
+inline std::vector<Elem> to_list(std::vector<IValue>& list) {
+  std::vector<Elem> converted_list;
+  std::transform(
+      list.begin(), list.end(), std::back_inserter(converted_list), [
+      ](IValue ivalue) -> Elem {
+        return ivalue.to<Elem>();
+      });
+  return std::move(converted_list);
+}
+
+template<typename Elem>
+inline std::vector<Elem> to_list(const std::vector<IValue>& list) {
+  std::vector<Elem> converted_list;
+  std::transform(
+      list.begin(), list.end(), std::back_inserter(converted_list), [
+      ](IValue ivalue) -> Elem {
+        return ivalue.to<Elem>();
+      });
+  return std::move(converted_list);
+}
+
+template<typename T>
+inline T IValue::to() && {
```

Comment:
We're hijacking the default template for `to()` to do only generic lists—what happens when someone wants to do `to<GenericDict>`?